src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -1722,13 +1723,13 @@ public object ForceVersionIncrement(object id, object currentVersion, ISessionIm
 		private SqlCommandInfo GenerateVersionIncrementUpdateString()
 		{
 			SqlUpdateBuilder update = new SqlUpdateBuilder(Factory.Dialect, Factory);
-			update.SetTableName(GetTableName(0));
+			update.SetTableName(RootTableName);
 			if (Factory.Settings.IsCommentsEnabled)
 			{
 				update.SetComment("forced version increment");
 			}
 			update.AddColumn(VersionColumnName, VersionType);
-			update.SetIdentityColumn(IdentifierColumnNames, IdentifierType);
+			update.SetIdentityColumn(RootTableIdentifierColumnNames, IdentifierType);
```

Comment:
Shouldn't it be changed to `RootTableKeyColumnNames` for consistency with `GenerateSelectVersionString`? (Or eventually the other way round, but I do not see which added value has `RootTableIdentifierColumnNames`. It is overridable contrary to `RootTableKeyColumnNames`, but never overridden in NHibernate code base.)

Reply:
`RootTableIdentifierColumnNames`

Reply:
There are also `SelectLockingStrategy` and `UpdateLockingStrategy`. I'm making everything consistent to `UpdateLockingStrategy`.